mm2src/coins/eth/eth_tests.rs

```diff
@@ -464,10 +496,12 @@ fn test_search_for_swap_tx_spend_was_spent() {
         &hex::decode("809465b17d0a4ddb3e4c69e8f23c2cabad868f51f8bed5c765ad1d6516c3306f").unwrap(),
     )
     .unwrap();
-    let transport = Web3Transport::new(vec![
-        "https://ropsten.infura.io/v3/c01c1b4cf66642528547624e1d6d9d6b".into()
-    ])
-    .unwrap();
+    let transport = Web3Transport::new(vec![Web3TransportNode {
+        uri: "https://ropsten.infura.io/v3/c01c1b4cf66642528547624e1d6d9d6b"
+            .parse()
+            .unwrap(),
+        gui_auth: false,
+    }]);
```

Comment:
What if we add `Web3Transport::for_test` that takes `uri: &str`?

Reply:
I thought it will be unnecessary because there will be tets that requires gui_auth as well. Also string-to-uri casting in `Web3Transport::new` function isn't really good idea since it's task simply creating `Web3Transport` not type casting. The function should't fail on invalid urls, therefore it should directly take the valid type instead of `&str`.

Reply:
Why not? If `Web3Transport::for_test` is marked as `#[cfg(test)]`, then it just simply panic.
> The function should't fail on invalid urls, therefore it should directly take the valid type instead of &str

Reply:
Sorry, I misunderstood. I thought I should make that change on `new` function.

> I thought it will be unnecessary because there will be tets that requires gui_auth as well. 

I guess we should make one function for gui_auth ones as well? But then, what if some nodes will support gui_auth and some not, in that case we will need to use `new` in that unit test. I am not sure if it's worth add such function.

Reply:
But if we only need single node in tests, then we can make it like `url: &'static str, gui_auth: bool`, and use `new` inside of that function. That way it can be actually pretty good. What do you think ? @sergeyboyko0791

Reply:
> But if we only need single node in tests, then we can make it like url: &'static str, gui_auth: bool, and use new inside of that function.

Yes, why not. I think it will be even better to add a global scope function `w3_transport_for_test` like we have for UTXO
https://github.com/KomodoPlatform/atomicDEX-API/blob/ae790660b0fc7350c7132bc7f0003454a8af4ea3/mm2src/coins/utxo/utxo_tests.rs#L48

Reply:
Created `single_node` fn without `test` attribute since this functionality is also needed in wasm tests. Can't use `wasm_bindgen_test` and `test` attributes together because one overrides the other.